tests/nested/core20/kernel-reseal/task.yaml

```diff
@@ -2,19 +2,37 @@ summary: Check that a kernel refresh reseals
 
 systems: [ubuntu-20.04-64]
 
-execute: |
+prepare: |
   # shellcheck source=tests/lib/nested.sh
   . "$TESTSLIB/nested.sh"
 
-  # Wait for snapd to be seeded
-  nested_exec sudo snap wait system seed.loaded
+  snap download pc-kernel --channel=20/edge --basename=pc-kernel
+  unsquashfs -d pc-kernel pc-kernel.snap
+  sed -i 's/This program cannot be run in DOS mode/This program cannot be run in D0S mode/' pc-kernel/kernel.efi
+
+  KEY_NAME=$(nested_get_snakeoil_key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+  # FIXME: this looks almost like nested_secboot_sign_gadget
```

Comment:
@bboozzoo proposed `nested_secboot_sign_file` in the gadget test pr which could be used here now

Reply:
Yeah, excellent point. Can add once that has landed